test/functional/specs/Data Collector/C1715149.js

```diff
@@ -0,0 +1,102 @@
+import { RequestLogger, t, ClientFunction } from "testcafe";
+import createNetworkLogger from "../../helpers/networkLogger";
+import createFixture from "../../helpers/createFixture";
+import { orgMainConfigMain } from "../../helpers/constants/configParts";
+import configureAlloyInstance from "../../helpers/configureAlloyInstance";
+
+const networkLogger = createNetworkLogger();
+
+const networkLoggerConfig = {
+  logRequestBody: true,
+  stringifyRequestBody: true
+};
+
+const destinationLogger = RequestLogger(
+  "https://cataas.com/cat",
+  networkLoggerConfig
+);
+
+const onBeforeEventSend = ClientFunction(() => {
+  window.onBeforeEventSendCalled = true;
+});
+
+const onBeforeEventSendFailed = ClientFunction(() => {
+  window.onBeforeEventSendCalled = true;
+  throw new Error("Expected Error");
+});
+
+const onBeforeEventSendFalse = ClientFunction(() => {
+  window.onBeforeEventSendCalled = true;
+  return false;
+});
+
+const getOnBeforeEventSendCalled = ClientFunction(() => {
+  return window.onBeforeEventSendCalled === true;
+});
+
+const getSendEventPromiseResult = ClientFunction(() => {
+  return window.sendEventPromiseResult;
+});
```

Comment:
The way you're configuring Alloy and sending events is "old school" as of a week or so ago. @jonsnyder did a bunch of work to make stuff like this easier. You'll need to merge the latest from `main` into your branch. Then check out how it's done in https://github.com/adobe/alloy/blob/main/test/functional/specs/Privacy/C2593.js#L24-L40

Also see Jon's documentation for `createAlloyProxy`: https://github.com/adobe/alloy/blob/d86b274002ab1c50d379b24b933bc938952de885/test/functional/helpers/createAlloyProxy.js#L2-L47